src/backend/dx11/src/conv.rs

```diff
@@ -26,60 +24,60 @@ pub fn map_index_type(ty: IndexType) -> DXGI_FORMAT {
 
 pub fn typeless_format(format: DXGI_FORMAT) -> Option<DXGI_FORMAT> {
     match format {
-        DXGI_FORMAT_R8G8B8A8_UNORM |
-        DXGI_FORMAT_R8G8B8A8_SNORM |
-        DXGI_FORMAT_R8G8B8A8_UINT |
-        DXGI_FORMAT_R8G8B8A8_SINT |
-        DXGI_FORMAT_R8G8B8A8_UNORM_SRGB => Some(DXGI_FORMAT_R8G8B8A8_TYPELESS),
+        DXGI_FORMAT_R8G8B8A8_UNORM
+        | DXGI_FORMAT_R8G8B8A8_SNORM
+        | DXGI_FORMAT_R8G8B8A8_UINT
+        | DXGI_FORMAT_R8G8B8A8_SINT
+        | DXGI_FORMAT_R8G8B8A8_UNORM_SRGB => Some(DXGI_FORMAT_R8G8B8A8_TYPELESS),
 
         // ?`
-        DXGI_FORMAT_B8G8R8A8_UNORM |
-        DXGI_FORMAT_B8G8R8A8_UNORM_SRGB => Some(DXGI_FORMAT_B8G8R8A8_TYPELESS),
-
-        DXGI_FORMAT_R8_UNORM |
-        DXGI_FORMAT_R8_SNORM |
-        DXGI_FORMAT_R8_UINT |
-        DXGI_FORMAT_R8_SINT => Some(DXGI_FORMAT_R8_TYPELESS),
-
-        DXGI_FORMAT_R8G8_UNORM |
-        DXGI_FORMAT_R8G8_SNORM |
-        DXGI_FORMAT_R8G8_UINT |
-        DXGI_FORMAT_R8G8_SINT => Some(DXGI_FORMAT_R8G8_TYPELESS),
-
-        DXGI_FORMAT_R16_UNORM |
-        DXGI_FORMAT_R16_SNORM |
-        DXGI_FORMAT_R16_UINT |
-        DXGI_FORMAT_R16_SINT |
-        DXGI_FORMAT_R16_FLOAT => Some(DXGI_FORMAT_R16_TYPELESS),
-
-        DXGI_FORMAT_R16G16_UNORM |
-        DXGI_FORMAT_R16G16_SNORM |
-        DXGI_FORMAT_R16G16_UINT |
-        DXGI_FORMAT_R16G16_SINT |
-        DXGI_FORMAT_R16G16_FLOAT => Some(DXGI_FORMAT_R16G16_TYPELESS),
-
-        DXGI_FORMAT_R16G16B16A16_UNORM |
-        DXGI_FORMAT_R16G16B16A16_SNORM |
-        DXGI_FORMAT_R16G16B16A16_UINT |
-        DXGI_FORMAT_R16G16B16A16_SINT |
-        DXGI_FORMAT_R16G16B16A16_FLOAT => Some(DXGI_FORMAT_R16G16B16A16_TYPELESS),
-
-        DXGI_FORMAT_D32_FLOAT |
-        DXGI_FORMAT_R32_UINT |
-        DXGI_FORMAT_R32_SINT |
-        DXGI_FORMAT_R32_FLOAT => Some(DXGI_FORMAT_R32_TYPELESS),
-
-        DXGI_FORMAT_R32G32_UINT |
-        DXGI_FORMAT_R32G32_SINT |
-        DXGI_FORMAT_R32G32_FLOAT => Some(DXGI_FORMAT_R32G32_TYPELESS),
-
-        DXGI_FORMAT_R32G32B32_UINT |
-        DXGI_FORMAT_R32G32B32_SINT |
-        DXGI_FORMAT_R32G32B32_FLOAT => Some(DXGI_FORMAT_R32G32B32_TYPELESS),
-
-        DXGI_FORMAT_R32G32B32A32_UINT |
-        DXGI_FORMAT_R32G32B32A32_SINT |
-        DXGI_FORMAT_R32G32B32A32_FLOAT => Some(DXGI_FORMAT_R32G32B32A32_TYPELESS),
+        DXGI_FORMAT_B8G8R8A8_UNORM | DXGI_FORMAT_B8G8R8A8_UNORM_SRGB => {
+            Some(DXGI_FORMAT_B8G8R8A8_TYPELESS)
+        }
+
+        DXGI_FORMAT_R8_UNORM | DXGI_FORMAT_R8_SNORM | DXGI_FORMAT_R8_UINT | DXGI_FORMAT_R8_SINT => {
+            Some(DXGI_FORMAT_R8_TYPELESS)
+        }
+
+        DXGI_FORMAT_R8G8_UNORM
+        | DXGI_FORMAT_R8G8_SNORM
+        | DXGI_FORMAT_R8G8_UINT
+        | DXGI_FORMAT_R8G8_SINT => Some(DXGI_FORMAT_R8G8_TYPELESS),
+
+        DXGI_FORMAT_R16_UNORM
+        | DXGI_FORMAT_R16_SNORM
```

Comment:
I find this controversial. It doesn't align with the first item, and it's not obvious to me what this improves

Reply:
While I agree it is not clearly better, there is no specific config option for this, `binop_seperator` which can be `"Front"` or `"Back"` however that effects all binary operators. This exact problem led to the allowance of a pipe at the start of a match clause, but I don't think that actually looks any better.